src/core/users/service/users.service.ts

```diff
@@ -0,0 +1,104 @@
+import { Injectable, NotFoundException, Param } from '@nestjs/common';
+import { CreateUserDto } from '../dto/create-user.dto';
+import { UpdateUserDto } from '../dto/update-user.dto';
+import { User } from '../entities/user.entity';
+import { Repository } from 'typeorm';
+import { Profile } from '../../profiles/entities/profile.entity';
+import { InjectRepository } from '@nestjs/typeorm';
+import * as bcryptjs from 'bcryptjs';
+import { JwtPayloadService } from '../../../common/service/jwt.payload.service';
+import { USER_NOT_FOUND } from '../messages';
+export class UsersService {
+  constructor(
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+
+    @InjectRepository(Profile)
+    private readonly profileRepository: Repository<Profile>,
+
+    private readonly jwtPayloadService: JwtPayloadService,
+  ) {}
+
+  async create(createUserDto: CreateUserDto) {
+    const user = await this.userRepository.create(createUserDto);
+
+    createUserDto.password = await bcryptjs.hash(createUserDto.password, 10);
+
+    await this.userRepository.save(user);
+
+    const getUser = await this.userRepository.findOne({
+      where: { email: createUserDto.email },
+    });
+
+    const perfil = await this.profileRepository.create({
+      userId: getUser.id,
+      description: 'default',
+      mainTitle: 'default',
+      countryResidence: 'default',
+    });
+    await this.profileRepository.save(perfil);
+
+    const token = await this.jwtPayloadService.createJwtPayload(user);
+
+    const response = {
+      user: getUser,
+      perfil,
+      token,
+    };
+
+    return response;
+  }
+
+  async findAll() {
+    return await this.userRepository.find();
+  }
+
+  async findOne(id: number) {
+    const user = await this.userRepository.findOneBy({ id });
+
+    if (!user) {
+      throw new NotFoundException(USER_NOT_FOUND);
+    }
+
+    return user;
+  }
+
+  async findOneByEmail(email: string) {
+    return await this.userRepository.findOneBy({ email });
+  }
+
+  async findByEmailWithPassword(email: string) {
+    return await this.userRepository.findOne({
+      where: { email },
+      select: [
+        'id',
+        'name',
+        'lastname',
+        'email',
+        'password',
+        'role',
+        'verified',
+      ],
+    });
+  }
+
+  async update(id: number, updateUserDto: UpdateUserDto): Promise<void> {
+    const user = await this.userRepository.update({ id }, updateUserDto);
+
+    if (user.affected === 0) {
+      throw new NotFoundException(USER_NOT_FOUND);
+    }
```

Comment:
No estoy seguro de que pase con las pre-validaciones, pero.... perfectamente podria fallar la BDD, seria bueno buscar una forma de interceptar errores de Prisma para que los manejes de alguna forma
```suggestion
    try {
        const user = await this.userRepository.update({ id }, updateUserDto);
        if (user.affected === 0) {
           throw new NotFoundException(USER_NOT_FOUND);
        }
        
    } catch (error) {
       ...
    }
```

De que forma?, quiza algo como (esto es netamente un ejemplo, no lo se a los identificadores de los errores de TypeOrm):

```suggestion
    try {
        const user = await this.userRepository.update({ id }, updateUserDto);
        if (user.affected === 0) {
           throw new NotFoundException(USER_NOT_FOUND);
        }
        
    } catch (error) {
       if (error instanceof TypeOrmError) throw new UnexpectedException(DATABASE_ERROR_MESSAGE);
       throw new error;
    }
```

Reply:
Ok hare mi research

Reply:
papax eviten esos try y catch que luegoa  typeorm le da por no enviar los mensajes de error del Throw new

Reply:
Es que esos try catch es la forma correcta de hacerlo @HopeAero xdxd, pero igual seria bueno que nos comentes que viste tu del por que no hacerlo para ver si es mejor en efecto, no hacerlos

Reply:
[Crear una nueva tarea llamada "Manejo de errores de TypeORM"]